types/src/system/auction/mod.rs

```diff
@@ -506,10 +506,6 @@ pub trait Auction:
         let base_round_reward = self.read_base_round_reward()?;
         let era_id = detail::get_era_id(self)?;
 
-        if reward_factors.keys().ne(seigniorage_recipients.keys()) {
-            return Err(Error::MismatchedEraValidators);
-        }
```

Comment:
Why are you removing this check? This was added after we had a situation/bug where those did indeed not match and node later panic'd.

Reply:
When we construct a block after genesis/upgrade we make the reward factors map empty.

Reply:
Then maybe we should be checking if it's a subset of the expected validator set? Rather than removing the check althogether. It's not strictly necessary but I think by removing it we're losing more that we could gain if we fit it into the new situation.

Reply:
If it's not a subset that will already cause a `ValidatorNotFound` error in line 515.